android/app/src/main/java/com/cBreez/client/BreezForegroundService.kt

```diff
@@ -0,0 +1,137 @@
+        )
+        Logger.tag(TAG).debug("Breez node service created.")
+    }
+
+    // =========================================================== //
+    //                      SERVICE LIFECYCLE                      //
+    // =========================================================== //
+
+    override fun onBind(intent: Intent?): IBinder {
+        Logger.tag(TAG).debug("Binding Breez node service from intent=$intent")
+        // UI is binding to the service. The service is not headless anymore and we can remove the notification.
+        isHeadless = false
+        stopForeground(STOP_FOREGROUND_REMOVE)
+        dismissForegroundServiceNotification(applicationContext)
+        return binder
+    }
+
+    /** When unbound, the service is running headless. */
+    override fun onUnbind(intent: Intent?): Boolean {
+        isHeadless = true
+        return false
+    }
+
+    private val shutdownHandler = Handler(Looper.getMainLooper())
+    private val shutdownRunnable: Runnable = Runnable {
+        if (isHeadless) {
+            Logger.tag(TAG).debug("Reached scheduled shutdown...")
+            stopForeground(STOP_FOREGROUND_DETACH)
+            shutdown()
+        }
+    }
+
+    /** Shutdown the node, close connections and stop the service */
+    private fun shutdown() {
+        Logger.tag(TAG).info("Shutting down Breez node service")
+        stopSelf()
+    }
+
+    // =========================================================== //
+    //                    START COMMAND HANDLER                    //
+    // =========================================================== //
+
+    /** Called when an intent is called for this service. */
+    @RequiresApi(Build.VERSION_CODES.TIRAMISU)
+    override fun onStartCommand(intent: Intent?, flags: Int, startId: Int): Int {
+        super.onStartCommand(intent, flags, startId)
+        Logger.tag(TAG)
+            .debug("Start Breez node service from intent [ intent=$intent, flag=$flags, startId=$startId ]")
+        startForeground(
+            NOTIFICATION_ID_FOREGROUND_SERVICE,
+            notifyForegroundService(applicationContext)
+        )
+        getParcelable(
+            intent, "remote_message",
+            RemoteMessage::class.java
+        )?.let { handleNow(it) }
+
+        shutdownHandler.removeCallbacksAndMessages(null)
+        shutdownHandler.postDelayed(shutdownRunnable, 60 * 1000L) // push back shutdown by 60s
+        if (!isHeadless) {
+            stopForeground(STOP_FOREGROUND_REMOVE)
+        }
+        return START_NOT_STICKY
+    }
+
+    private fun handleNow(remoteMessage: RemoteMessage): Boolean {
+        return if (remoteMessage.data["notification_type"] == "payment_received") {
+            val paymentHash = remoteMessage.data["payment_hash"]
+            val clickAction = remoteMessage.data["click_action"]
+            paymentHash?.let {
+                JobManager.instance.startPaymentReceivedJob(
```

Comment:
I don't think we need a job here as we already running in a foreground service, we should have cpu and time.

Reply:
[JobManager](https://github.com/breez/c-breez/blob/main/android/app/src/main/java/com/cBreez/client/JobManager.kt
) is just a proxy to enqueue work requests.

From Android 14 behavior changes: [Foreground service types are required](https://developer.android.com/about/versions/14/changes/fgs-types-required)

> If a use case in your app isn't associated with any of these types, we strongly recommend that you migrate your logic to use [WorkManager](https://developer.android.com/topic/libraries/architecture/workmanager) or [user-initiated data transfer jobs](https://developer.android.com/about/versions/14/changes/user-initiated-data-transfers).

Reply:
I don't understand why we need to use work manager here: https://github.com/breez/c-breez/blob/main/android/app/src/main/java/com/cBreez/client/JobManager.kt#L60

If we are using a service shouldn't we just run the code as we already get the cpu and time we need?

Reply:
> I suppose type of our foreground service would be considered [`dataSync`](https://developer.android.com/about/versions/14/changes/fgs-types-required#data-sync), which will be also be deprecated and WorkManager is the most fitting of the suggested alternatives.
> 
> We could decide whether if there's a limiting factor in using [`shortService`](https://developer.android.com/about/versions/14/changes/fgs-types-required#short-service) type for our job and forgo WorkManager.

I don't think dataSync is the right type as dataSync is something that happens periodically and not critical. It seems that shortService is more suitable.